libs/@hashintel/component-library/src/lib/flexible-filter.tsx

```diff
@@ -0,0 +1,483 @@
+import type { ImageData } from "canvas";
```

Comment:
don't these need the `"use client"` pragma because they can only be used on the client side and not be rendered during RSC? (primarily because of the `useEffect`)

Reply:
I don't think any of our current apps use RSC _yet_, so I don't think we have one for it. This will probably be something we'll just have to do one by one. Just a flag that with RSC it'd likely be a requirement.

> fun fact: react router does have experimental support for RSC - https://reactrouter.com/how-to/react-server-components